Automattic/sensei
# Student progress API

Closes https://github.com/Automattic/sensei/issues/5407

### Changes proposed in this Pull Request

* Add new Student Progress API.
* Replace some usages of old functions/methods with new ones.

### Testing instructions

* Create a few courses with lessons (with quizzes and without them).
* Create a few students.
* Try to take some courses yourself, complete lessons and quizzes.
* Check your progress in admin panel is correct.
* Try to add/remove students to courses, etc.
* No errors expected, no changes in behaviour expected.
